pipenv/project.py

```diff
@@ -1111,12 +1111,24 @@ def get_package_name_in_pipfile(self, package_name, category):
                 return name
         return None
 
+    def _sort_category(self, category):
+        # toml tables won't maintain sorted dictionary order
+        # so construct the table in the order that we need
+        sorted_category = dict(sorted(category.items()))
```

Comment:
I think this will definitely work, but I am wondering if it would be more efficient to move into the loop and  not cast to a dict so like:
`for key, value in sorted(category.items():`

Correct me if that won't work or there is an issue with it, but thats my last bit of feedback.   I'll see if @oz123 can review this too but I want to get this merged into the release for this month.

Reply:
Thank you very much! That is very kind.

Are there other test cases I should consider or other tests I can add?

I noticed some projects that have `# comments` on some lines of their Pipfiles. When I add a comment on both string-type lines and dictionary-type lines those are both sorted correctly.

Are there other values or types that can go on a line?

Reply:
> I am wondering if it would be more efficient to move into the loop and not cast to a dict so like: `for key, value in sorted(category.items():`

Thanks! I created a PR to do that - https://github.com/pypa/pipenv/pull/5990/files

Is it helpful if I create a news fragment `.rst` ? Or is that no longer useful since the PR has been merged?

Reply:
the new change is trivial and the docs change you did is good -- we just need to figure out why the docs don't seem to be getting published at some point during the new releases.